syntax="proto3";

package user.error;

enum UserErrorCode {
    ERROR_UNSPECIFIED = 0;
    ERROR_INVALID_PARAMETER = 1;
    ERROR_USER_NOT_FOUND = 2;
    ERROR_INVALID_ENTITY_TYPE = 3;
    ERROR_ENTITY_USER_NOT_FOUND = 4;
    ERROR_INVALID_PERMISSION_ROLE = 5;
    ERROR_OPERATION_FAILED = 6;
    ERROR_EMAIL_ALREADY_REGISTERED = 10;
    ERROR_USER_EMAIL_NOT_FOUND = 250;
    ERROR_USER_EMAIL_NOT_VERIFIED = 251;
    ERROR_USER_EMAIL_NOT_UNIQUE = 252;
    ERROR_INVALID_USER_ATTRIBUTES = 242;
}
